amazon-kinesis-client-multilang/src/main/java/software/amazon/kinesis/multilang/config/AwsCredentialsProviderPropertyValueDecoder.java

```diff
@@ -63,25 +67,25 @@ public AWSCredentialsProvider decodeValue(String value) {
      * @return list of supported types
      */
     @Override
-    public List<Class<AWSCredentialsProvider>> getSupportedTypes() {
-        return Collections.singletonList(AWSCredentialsProvider.class);
+    public List<Class<AwsCredentialsProvider>> getSupportedTypes() {
+        return Collections.singletonList(AwsCredentialsProvider.class);
     }
 
     /**
      * Convert string list to a list of valid credentials providers.
      */
-    private static List<AWSCredentialsProvider> getValidCredentialsProviders(List<String> providerNames) {
-        List<AWSCredentialsProvider> credentialsProviders = new ArrayList<>();
+    private static List<AwsCredentialsProvider> getValidCredentialsProviders(List<String> providerNames) {
+        List<AwsCredentialsProvider> credentialsProviders = new ArrayList<>();
 
         for (String providerName : providerNames) {
             final String[] nameAndArgs = providerName.split("\\" + ARG_DELIMITER);
-            final Class<? extends AWSCredentialsProvider> clazz;
+            final Class<? extends AwsCredentialsProvider> clazz;
             try {
                 final Class<?> c = Class.forName(nameAndArgs[0]);
-                if (!AWSCredentialsProvider.class.isAssignableFrom(c)) {
+                if (!AwsCredentialsProvider.class.isAssignableFrom(c)) {
                     continue;
                 }
-                clazz = (Class<? extends AWSCredentialsProvider>) c;
+                clazz = (Class<? extends AwsCredentialsProvider>) c;
```

Comment:
This will break backward compatibility. The method tries to derived from the real class based on the provided name.  Using the new `KclStsAssumeRoleCredentialsProvider` for example, the old class uses capital letter `STS` so if the for customer that still uses the old STS provider name in their property file this will not be able to resolve to the new class. This applies to other aws credential providers as well.

We might have to come up with some mapping ourself based on this list otherwise we need a multilang revbump https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/migration-client-credentials.html

Reply:
I think we'll need to have a conversation about whether we still want to support v1 credentials providers. 

This will require quite a bit more work to ensure it works for both provider types. Previously, we were converting all v1 providers to v2. If we are supporting a mix of v1 and v2 providers, we need to rethink how we are wrapping these / separating these into different processes.

Reply:
We have decided to move forward without backwards compatibility to v1.